src/clib/Makefile

```
@@ -198,56 +200,35 @@ verbose: VERBOSE = 1
 #-----------------------------------------------------------------------
 
 .PHONY: autogen
-autogen: config_type auto_show_config.c auto_show_flags.c auto_get_version.c
+autogen: config_type auto_general.c
 
 .PHONY: config_type
-config_type:
-	-@echo "#ifndef __GRACKLE_FLOAT_H__" > grackle_float.h
-	-@echo "#define __GRACKLE_FLOAT_H__" >> grackle_float.h
-	-@echo "#define $(PRECISION_DEFINE)" >> grackle_float.h
-	-@echo "#endif" >> grackle_float.h
-
-# Force update of auto_show_config.c
+config_type: grackle_float.h.in
+	@(if [ $(ASSEMBLE_PRECISION_NUMBER) -eq 4 ]; then \
+	   $(CONFIG_DIR)/configure_file.py --clobber \
+	       --input grackle_float.h.in \
+	       --output grackle_float.h \
+	       GRACKLE_FLOAT_MACRO=GRACKLE_FLOAT_4; \
+	else \
+	   $(CONFIG_DIR)/configure_file.py --clobber \
+	       --input grackle_float.h.in \
+	       --output grackle_float.h \
+	       GRACKLE_FLOAT_MACRO=GRACKLE_FLOAT_8; \
+	fi)
```

Maybe without the if statement like this?
```suggestion
	@$(CONFIG_DIR)/configure_file.py --clobber \
		--input grackle_float.h.in \
		--output grackle_float.h \
		GRACKLE_FLOAT_MACRO=GRACKLE_FLOAT_$(ASSEMBLE_PRECISION_NUMBER);
```

---

Good idea. Fixed